docs/source/contributing/contributing.md

```diff
@@ -1,45 +1,86 @@
+    Feature requests are a great opportunity for you to advocate for the use case you're suggesting.
+    They help others understand how much effort it would be to integrate the work,and - if you're successful at convincing them that this effort is worth it - make it more likely that they to choose to work on it with you.
+
+2. Open an issue.
+  Getting consensus with the community is a great way to save time later.
+3. Make edits in [your fork](https://help.github.com/en/articles/fork-a-repo) of the [repo2docker repository](https://github.com/jupyter/repo2docker).
+4. Make a [pull request](https://help.github.com/en/articles/about-pull-requests).
+Read the [next section](#guidelines-to-getting-a-pull-request-merged) for guidelines for both reviewers and contributors on merging a PR.
+5. Edit [the changelog](./../../changelog) by appending your feature / bug fix to the development version.
+6. Wait for a community member to merge your changes.
+7. (optional) Deploy a new version of repo2docker to mybinder.org by [following these steps](http://mybinder-sre.readthedocs.io/en/latest/deployment/how.html)
+
+## Guidelines to getting a Pull Request merged
 
+These are not hard rules to be enforced by 🚓 but they are suggestions written by the repo2docker maintainers to make getting your contribution completed a smooth process for you and for them.
+
+* **Create a PR as early as possible**, marking it with `[WIP]` while you work on it.
+  This avoids duplicated work, lets you get high level feedback on functionality or API changes, and/or helps find collaborators to work with you.
+* **Keep your PR focused.**
+  The best PRs solve one problem.
+  If you end up changing multiple things, please open separate PRs for the different conceptual changes.
+* **Add tests your code.**
+  PRs will not be merged if Travis is failing.
+* **Apply [PEP8](https://www.python.org/dev/peps/pep-0008/)** as much as possible, but not too much.
+  If in doubt, ask.
+* **Use merge commits** instead of merge-by-squashing/-rebasing.
+  This makes it easier to find all changes since the last deployment `git log --merges --pretty=format:"%h %<(10,trunc)%an %<(15)%ar %s" <deployed-revision>..` and your PR easier to review.
+* **Make it clear when your PR is ready for review.**
+  Prefix the title of your pull request (PR) with `[MRG]` if the contribution is complete and should be subjected to a detailed review.
+* **Enter your changes into the [changelog](./../../changelog)** in `docs/source/changelog.rst`.
+* **Use commit messages to describe _why_ you are proposing the changes you are proposing.**
+* **Try to not rush changes** (the definition of rush depends on how big your changes are).
+  Remember that everyone in the repo2docker team is a volunteer and we can not (nor would we want to) control their time or interests.
+  Wait patiently for a reviewer to merge the PR.
```

Comment:
Related to this: the guidelines are for both maintainers and contributors. So the "don't rush" also applies to reviewers, it is meant to remind people that other timezones might be sleeping right now and waiting 24h to give them a chance to see, think and comment is a good way to avoid them being surprised by changes already in master.

In similar fashion: ` * someone else has to merge your PR;` got lost in the move and is (maybe) the only hard rule I am absolutely convinced of enforcing. What do you think about duplicating/splitting the guide into "for contributors" and "for maintainers"? My feeling is that there is too much overlap to make it worth it -> include this in this part of the guide.

Reply:
Heya! I've added a line on 66 and 92 (just below here). What do you think?

I was trying _not_ to have a separation of contributors and matinainers, for the reason that we discussed somewhere else in this thread - that often they're the _same people_ just wearing different hats.

I think we could built up a guide for reviewers if that seems useful? But my feeling at the moment is that its just another lens on the same information (so I agree with you that there's a lot of overlap).

Reply:
Looks good!